### ISSUE TITLE
Feature: CommandSender in DynamicSuggestedStringArgument

### ISSUE THREAD

**Issue by Draycia** [JorelAli/1.13-Command-API#41]
**Please describe your suggestion**
Would be nice to dynamically suggest strings per user. For instance, each user has a set of homes, and I would like to display their homes as they're typing the argument.

**Please supply examples of the desired inputs and outputs**
_Example:_
Input: `/home homename`
Output: `/home <home, farm, friend, newbase>`

_Example:_
```kotlin
class CommandHome {
    private fun run(sender: CommandSender, args: Array<Any>) {
        if (sender !is Player) return;

        // Check if home exists and teleport player there.
    }

    private fun dynamicStringArgument(sender: CommandSender): Array<String> {
        if (sender !is Player) return arrayOf();
        return PluginMain.homes[sender.uniqueId] ?: arrayOf();
    }

    fun registerCommand() {
        val normal = LinkedHashMap<String, Argument>();
        normal["Home"] = DynamicSuggestedStringArgument(::dynamicStringArgument);
        CommandAPI.getInstance().register("home", CommandPermission.fromString("commands.home"), normal, ::run);

    }
}
```

**Comment by JorelAli** [comment]
Sounds doable. I'll take a look at it!

**Comment by Draycia** [comment]
Thank you!

**Comment by JorelAli** [comment]
Implemented in [this commit](https://github.com/JorelAli/1.13-Command-API/commit/bb50a942831e7cfd576eed662325f0638739d94c)

This will be included in the 1.9 release which is due soon!
